near-sdk/src/json_types/public_key.rs

```diff
@@ -5,7 +5,17 @@ use std::convert::TryFrom;
 
 /// PublicKey curve
 #[derive(
-    Debug, Clone, Copy, Serialize, Deserialize, PartialEq, BorshDeserialize, BorshSerialize,
+    Debug,
+    Clone,
+    Copy,
+    Serialize,
+    Deserialize,
+    PartialOrd,
+    Ord,
+    Eq,
+    PartialEq,
+    BorshDeserialize,
+    BorshSerialize,
```

Comment:
I'm actually not sure why there are serialize impls on this, it probably shouldn't as it's inconsistent and would be a foot gun. For the `Base58PublicKey` it should have a custom borsh implementation so these shouldn't be used at all

Reply:
Maybe they were there before borsh was a thing? In any case, I'll try and remove them

Reply:
I mean, the type is exported publicly now, so probably best to either make it only visible within the crate or have a function to return a reference to the curve type on the public key type. Either way I don't see the need for serialize impls (either) to be implemented for it. There _shouldn't_ be a need to serialize just the curve type.

Reply:
mmm, I'd imagine the developer potentially wanting to access the curve type, so I think having the function to a reference of it would be best. Well, this is pretty much going to be a breaking change already, so getting rid Serialize impls for both `CurveType` and `Base58PublicKey` should be fine, right?

Reply:
no keep for `Base58PublicKey`, that is the whole point of this type is serialization. Get rid of it for `CurveType` though imo.